core/metacling/src/TClingMethodInfo.cxx

```diff
@@ -265,8 +265,11 @@ TClingMethodInfo::TClingMethodInfo(cling::Interpreter *interp,
 
       // Assemble special functions (or FunctionTemplate-s) that are synthesized from DefinitionData but
       // won't be enumerated as part of decls_begin()/decls_end().
-      auto Ctors = SemaRef.LookupConstructors(CXXRD);
-      SpecFuncs.append(Ctors.begin(), Ctors.end());
+      for (clang::NamedDecl *ctor : SemaRef.LookupConstructors(CXXRD)) {
+         // Filter out constructor templates, they are not functions we can iterate over:
+         if (auto *CXXMD = llvm::dyn_cast<clang::CXXMethodDecl>(ctor))
```

Comment:
```suggestion
         if (auto *CXXMD = llvm::dyn_cast<clang::CXXConstructorDecl>(ctor))
```

Reply:
> or maybe we also need CXXConversionDecl.

Conversion functions are not constructors (they create the return type, not their class type).

> for reference what is the Clang type of  "constructor templates"

`FunctionTemplateDecl`, unsurprisingly